packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -63,18 +63,19 @@ const App = (props) => {
     const location = useLocation()
     const authModal = useAuthModal()
     const customer = useCustomer()
+    const appConfig = useAppConfig()
 
-    const site = useSite()
-    const locale = useLocale()
+    const {pathname, search} = useLocation()
+    const site = useMemo(() => {
+        return resolveSiteFromUrl(`${pathname}${search}`)
+    }, [pathname, search])
+    const locale = useMemo(() => {
```

Comment:
There's already `targetLocale` from the App's `props`. And it's calculated from `resolveLocaleFromUrl` too.

Reply:
The App component uses `locale.preferredCurrency` besides `targetLocale` so we still need to retrieve the `preferredCurrency` by using `resolveLocaleFromUrl`:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/13f6523eff6869240da08903981e333abba4e15a/packages/template-retail-react-app/app/components/_app/index.jsx#L86

It looks `targetLocale` is obtained in getProps via a different method `getTargetLocale`:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/13f6523eff6869240da08903981e333abba4e15a/packages/template-retail-react-app/app/components/_app/index.jsx#L316